Keen/Query/Queries.cs

```diff
@@ -2,66 +2,93 @@
 using Newtonsoft.Json.Linq;
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
-using System.Dynamic;
-using System.IO;
 using System.Linq;
-using System.Net.Http;
-using System.Text;
 using System.Threading.Tasks;
 
+
 namespace Keen.Core.Query
 {
     /// <summary>
     /// Queries implements the IQueries interface which represents the Keen.IO Query API methods.
     /// </summary>
     internal class Queries : IQueries
     {
-        private IProjectSettings _prjSettings;
-        private string _serverUrl;
+        private readonly IKeenHttpClient _keenHttpClient;
+        private readonly string _queryRelativeUrl;
+        private readonly string _key;
```

Comment:
As I look at it now, after investigating Access Keys a bit more, I think those will be an entirely different API surface that leverages `KeenHttpClient` or an intermediate layer of abstraction. One shouldn't/needn't create a `KeenClient` instance until the appropriate keys are generated. `AccessKeys` will only require the `MasterKey`, and by whatever mechanism that's be acquired, a `KeenClient` can be created. Seems like those keys and the `KeenUrl` ought not change after that. If those keys need to change, recreating the instance won't be the end of the world.